src/Operation/Count.php

```diff
@@ -162,6 +162,39 @@ public function execute(Server $server)
         return (integer) $result->n;
     }
 
+    /**
+     * Explain the operation.
+     *
+     * @see Explainable::explain()
+     * @param Server $server
+     * @param $command
+     * @param array $options
+     * @return array|object
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function explain($server, $command, $options = [])
+    {
+        if ( ! isset($options['verbosity'])) {
+            $options['verbosity'] = 'allPlansExecution';
+        }
+
+        $cmd = new Command(['explain' => ['count' => $this->collectionName, 'query' => $command], 'verbosity' => $options['verbosity']]);
+
+        if (empty($command)) {
+            $cmd = new Command(['explain' => ['count' => $this->collectionName], 'verbosity' => $options['verbosity']]);
+        }
+
+        $result = $server->executeCommand($this->databaseName, $cmd);
+
+        $resultArray = get_object_vars($result->toArray()[0]); // cast $result to array
```

Comment:
Applying a type map of `['root' => 'array']` to the cursor (`$result` in this case) would be the more conventional way to work with an array here, as that allows the driver to do the conversion on its own and forego this object-to-array cast.

That said, if we want to give users control over the returned document, we should solicit a `typeMap` option and use that to apply to the array. If we think of the Explain operation as returning only a single document, it's quite similar to FindOne in that respect. With FindOne, the `typeMap` option gets applied to the Find operation's cursor, and FindOne just unpacks and returns the first and only result document in that cursor.

Also a reminder that PHPLIB has its own default type maps (BSONDocument and BSONArray classes), so even though it may be uncommon for users to specify their own type maps, we use this option to keep the various operations consistent. That's why operations such as CreateCollection take a typeMap option, even though they return a raw command result.